functorch/functorch/_src/aot_autograd.py

```diff
@@ -174,146 +175,192 @@ def call_func_with_args(f, args, steal_args=False):
     if not steal_args:
         args = list(args)
     assert isinstance(args, list)
-    # TODO: Please remove soon
-    # https://github.com/pytorch/pytorch/pull/83137#issuecomment-1211320670
+
+    # Disable the JIT Autocast flag to prevent re-autocasting of jitted graph.
+    # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
+    old_jit_autocast_flag = torch._C._jit_set_autocast_mode(False)
```

Comment:
this should be a try/catch or a context manager, if something fails before restoring the autocast mode state then the function is left in a weird state

Reply:
We're worried about overhead of a context manager. Also, I think the case we're catching is pretty unusual, and would require a pretty unusual confluence of situations to matter (torchscript + aotautograd in same program, amp, aotautograd errors inside call but we want to recover, etc.)

Reply:
Actually, seems like you only need this during compilation.